### DISCUSSION TITLE
Multiple Output types

### DISCUSSION THREAD

**Discussion by brittonhayes** [jdxyw/generativeart#4] [Ideas]
## Idea :bulb: 

Hey there :wave: ! Love this project and have been enjoying playing around with it. I thought it could be really nice if it was possible to output the result of the generated art as a `[]byte`. This would be in addition to the currently available options of saving a JPEG or PNG into the user's provided file path.

### c.ToBytes()

This shows a super simple example of the c.ToBytes() idea. It's a pretty minor addition to your package, but it would allow people to use the generative art package in a few new contexts. 

#### TLDR

I would love to respond to an `http` request with the generated art. A `ToBytes()` method would enable this.

```go
// Basic Usage

func DrawMyBytes() []byte {
    rand.Seed(time.Now().Unix())
    c := generativeart.NewCanva(500, 500)
    c.SetBackground(common.Black)
    c.FillBackground()
    c.SetColorSchema(common.DarkRed)
    c.SetForeground(common.LightPink)
    c.Draw(arts.NewJanus(10, 0.2))

    // Here is the new method I'm proposing
    // for the *generativeart.Canva type
    return c.ToBytes()
}
```

```go
// Using the above example function in an http context

func (h HttpHandler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
    // Draw the art as a slice of bytes
    b := DrawMyBytes()

    // Write the bytes to the http.ResponseWriter
    res.Write(b)
}
```

Would be interested to hear your thoughts on this!

**Comment by jdxyw** [comment]
sounds interesting. If you are interested in this project, you are welcome to submit a PR for it:)

**Reply by brittonhayes** [reply]
I'll take a look today and see if I can implement a ToBytes() method. Your code is super clean so I don't see it being too hard to add another method. 😄 

I'll report back if I run into any issues though!
